pkg/resources/external_table.go

```diff
@@ -0,0 +1,332 @@
+	},
+	"refresh_on_create": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Specifies to retain the access permissions from the original table when an external table is recreated using the CREATE OR REPLACE TABLE variant",
+		Default:     true,
+		ForceNew:    true,
+	},
+	"auto_refresh": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Specifies whether to automatically refresh the external table metadata once, immediately after the external table is created.",
+		Default:     true,
+		ForceNew:    true,
+	},
+	"copy_grants": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Specifies to retain the access permissions from the original table when an external table is recreated using the CREATE OR REPLACE TABLE variant",
+		Default:     false,
+		ForceNew:    true,
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		ForceNew:    true,
+		Description: "Specifies a comment for the external table.",
+	},
+	"owner": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		ForceNew:    true,
+		Description: "Name of the role that owns the external table.",
+	},
+}
+
+func ExternalTable() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateExternalTable,
+		Read:   ReadExternalTable,
+		Delete: DeleteExternalTable,
+		Exists: ExternalTableExists,
+
+		Schema: externalTableSchema,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+type externalTableID struct {
+	DatabaseName      string
+	SchemaName        string
+	ExternalTableName string
+}
+
+//String() takes in a externalTableID object and returns a pipe-delimited string:
+//DatabaseName|SchemaName|ExternalTableName
+func (si *externalTableID) String() (string, error) {
+	var buf bytes.Buffer
+	csvWriter := csv.NewWriter(&buf)
+	csvWriter.Comma = externalTableIDDelimiter
+	dataIdentifiers := [][]string{{si.DatabaseName, si.SchemaName, si.ExternalTableName}}
+	err := csvWriter.WriteAll(dataIdentifiers)
+	if err != nil {
+		return "", err
+	}
+	strExternalTableID := strings.TrimSpace(buf.String())
+	return strExternalTableID, nil
+}
+
+// externalTableIDFromString() takes in a pipe-delimited string: DatabaseName|SchemaName|ExternalTableName
+// and returns a externalTableID object
+func externalTableIDFromString(stringID string) (*externalTableID, error) {
+	reader := csv.NewReader(strings.NewReader(stringID))
+	reader.Comma = externalTableIDDelimiter
+	lines, err := reader.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("Not CSV compatible")
+	}
+
+	if len(lines) != 1 {
+		return nil, fmt.Errorf("1 line at a time")
+	}
+	if len(lines[0]) != 3 {
+		return nil, fmt.Errorf("3 fields allowed")
+	}
+
+	externalTableResult := &externalTableID{
+		DatabaseName:      lines[0][0],
+		SchemaName:        lines[0][1],
+		ExternalTableName: lines[0][2],
+	}
+	return externalTableResult, nil
+}
```

Comment:
I know that there are several copies of this code already, but at some point we should probably stop copying it and instead create something reusable.